prow/crier/reporters/gerrit/reporter.go

```diff
@@ -156,39 +155,38 @@ func (c *Client) Report(pj *v1.ProwJob) ([]*v1.ProwJob, error) {
 	pjTypeLabel := kube.ProwJobTypeLabel
 	gerritReportLabel := client.GerritReportLabel
 
-	selector := labels.Set{
-		clientGerritRevision: pj.ObjectMeta.Labels[clientGerritRevision],
-		pjTypeLabel:          pj.ObjectMeta.Labels[pjTypeLabel],
-	}
-	if pj.ObjectMeta.Labels[gerritReportLabel] == "" {
-		// Shouldn't happen, adapter should already have defaulted to Code-Review
-		logger.Errorf("Gerrit report label not set for job %s", pj.Spec.Job)
-	} else {
-		selector[gerritReportLabel] = pj.ObjectMeta.Labels[gerritReportLabel]
-	}
+	var toReportJobs []*v1.ProwJob
+	if pj.ObjectMeta.Labels[gerritReportLabel] == "" && pj.Status.State != v1.AbortedState {
+		toReportJobs = append(toReportJobs, pj)
+	} else { // generate an aggregated report
 
-	// list all prowjobs in the patchset matching pj's type (pre- or post-submit)
+		// list all prowjobs in the patchset matching pj's type (pre- or post-submit)
 
-	pjsOnRevisionWithSameLabel, err := c.lister.List(selector.AsSelector())
-	if err != nil {
-		logger.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
-		return nil, err
-	}
+		selector := labels.Set{
+			clientGerritRevision: pj.ObjectMeta.Labels[clientGerritRevision],
+			pjTypeLabel:          pj.ObjectMeta.Labels[pjTypeLabel],
+			gerritReportLabel:    pj.ObjectMeta.Labels[gerritReportLabel],
+		}
 
-	// generate an aggregated report:
-	var toReportJobs []*v1.ProwJob
-	mostRecentJob := map[string]*v1.ProwJob{}
-	for _, pjOnRevisionWithSameLabel := range pjsOnRevisionWithSameLabel {
-		job, ok := mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job]
-		if !ok || job.CreationTimestamp.Time.Before(pjOnRevisionWithSameLabel.CreationTimestamp.Time) {
-			mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job] = pjOnRevisionWithSameLabel
+		pjsOnRevisionWithSameLabel, err := c.lister.List(selector.AsSelector())
+		if err != nil {
+			logger.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
+			return nil, err
 		}
-	}
-	for _, pjOnRevisionWithSameLabel := range mostRecentJob {
-		if pjOnRevisionWithSameLabel.Status.State == v1.AbortedState {
-			continue
+
+		mostRecentJob := map[string]*v1.ProwJob{}
+		for _, pjOnRevisionWithSameLabel := range pjsOnRevisionWithSameLabel {
+			job, ok := mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job]
+			if !ok || job.CreationTimestamp.Time.Before(pjOnRevisionWithSameLabel.CreationTimestamp.Time) {
+				mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job] = pjOnRevisionWithSameLabel
+			}
+		}
+		for _, pjOnRevisionWithSameLabel := range mostRecentJob {
+			if pjOnRevisionWithSameLabel.Status.State == v1.AbortedState {
+				continue
```

Comment:
nit: It seems strange to filter out aborted PJs __after__ selecting the most recent PJ for each job name. If there are two PJs for a given job name and the later is aborted we should still report the results of the earlier unaborted job.

Reply:
We also shouldn't be doing any PJ dedupping at all here. Aborting duplicated jobs is already handled by the execution agent (e.g. `plank`) so the crier reporter can just report any (non-aborted) jobs it sees.

Reply:
hmm interesting, yeah I think the UX for that case can be improved (by reporting the older job). 
We still need to dedup finished jobs till they get gc'ed right?

Reply:
If we already reported a finished job then Crier should have marked the last report state as "finished", right? We should use that mechanism to determine if we need to report a PJ instead of assuming that the most recent PJ is the right one (and the only one) to report.